Leanplum-SDK/Classes/LPMessageTemplates.m

```diff
@@ -1025,6 +1030,26 @@ - (void)updatePopupLayout
     }
 }
 
+/**
+ * Get float value by parsing the html string that can have either % or px as a suffix.
+ */
+- (CGFloat)valueFromHtmlString:(NSString *)htmlString percentRange:(CGFloat)percentRange
+{
+    if (!htmlString || [htmlString length] == 0) {
+        return 0;
+    }
+
+    if ([htmlString hasSuffix:@"%"]) {
+        NSString *percentageValue = [htmlString stringByReplacingOccurrencesOfString:@"%"
+                                                                          withString:@""];
+        return percentRange * [percentageValue floatValue] / 100.;
+    }
+    
+    NSCharacterSet *letterSet = [NSCharacterSet letterCharacterSet];
+    NSArray *components = [htmlString componentsSeparatedByCharactersInSet:letterSet];
```

Comment:
could components be nil ?

Reply:
Since the htmlString is not nil, it will always return an array.

Reply:
Since the htmlString is not nil, it will always return an array.